### ISSUE TITLE
Frontend locale menu doesn't work on login form (Security) 

### ISSUE THREAD

**Issue by purplespider** [tractorcow-farm/silverstripe-fluent#689]
The locale menu (`LocaleMenu.ss`) doesn't appear to work when on the login form or password reset form (e.g. `Security`).

PR #688 fixes the issue with actions not being included in links, but there is a 2nd issue that results in the links for all Locales within the locale menu pointing to the "current" locale, instead of to the other relevant locale.

I believe the issue is `getRecordLocale()`([Link](https://github.com/tractorcow-farm/silverstripe-fluent/blob/5.0/src/Extension/FluentExtension.php#L813)) is failing to obtain the relevant local when on `Security` and is falling back to `getCurrentLocale()`.

But I can't get my head around how it could be resolved.

_Fluent v5.0.0_

**Comment by tractorcow** [comment]
it SHOULD work, but it works with the locale querystring argument, instead of the url prefix.

I.e. `Security/login?l=en_US`

This is due to `Security` being a controller, and not a page.

If it's not doing that, then it should be fixed to do that. :D

**Comment by tractorcow** [comment]
Here's the code in question.

https://github.com/tractorcow-farm/silverstripe-fluent/blob/5.0/src/Extension/FluentSiteTreeExtension.php#L109-L116

**Comment by tractorcow** [comment]
> I believe the issue is getRecordLocale()(Link) is failing to obtain the relevant local when on Security and is falling back to getCurrentLocale().

Sorry, you are right, it's probably that code in question failing on unsaved records. :)

**Comment by purplespider** [comment]
Yep, it's the generation of the locale menu items that has the issue.

**Default LocaleMenu.ss**
```html
            <ul>
                <% loop $Locales %>
                    <li class="$LinkingMode">
                        <a href="$Link.ATT" <% if $LinkingMode != 'invalid' %>rel="alternate"
                           hreflang="$HrefLang"<% end_if %>>$Title.XML</a>
                    </li>
                <% end_loop %>
            </ul>
```

**Results in:** From login screen. (once #688 is resolved)
```html
<ul>
        <li class="current">
            <a href="/Security/login?l=de_AT" rel="alternate"
                hreflang="de-at">German</a>
        </li>
    
        <li class="link">
            <a href="/Security/login?l=de_AT" rel="alternate"
                hreflang="en-gb">English</a>
        </li>
</ul>
```
(Note that the "English" link has the incorrect "de_AT" query string argument, instead of "en_GB")

**Comment by tractorcow** [comment]
Possible fix at https://github.com/tractorcow-farm/silverstripe-fluent/pull/691


The issue is that RecordLocale doesn't tell "unsaved" records what it's locale is. We can clone and mock the query param for that object to force it. :)
